### PR TITLE
feat: made stream a bit more generic

### PR DESCRIPTION
**Pull request by tdejager**
Made the stream function a bit more generic.

### CODE REVIEW

File: src/repror/internals/commands.py [hunk 1]
```diff
@@ -8,32 +8,51 @@
...
-        process.stderr, encoding="utf-8"
-    ):  # or another encoding
-        print(line, end="")
 
-    # Also print stderr if there is any
-    print(process.stdout.readline())
+    if stream_type == StreamType.STDERR:
+        kwarwgs = {"stderr": subprocess.PIPE}
+    else:
+        kwarwgs = {"stdout": subprocess.PIPE}
```
Comment (nichmor) [inline src/repror/internals/commands.py:40]: kwargs = {stderr: } if stream_type.is_stderr else {stdout }

File: src/repror/internals/commands.py [hunk 2]
```diff
@@ -8,32 +8,51 @@
...
     """Run a specific command."""
     return subprocess.run(command, cwd=cwd, env=env, check=True, capture_output=silent)
 
 
-def run_streaming_command_stderr(
+class StreamType(Enum):
+    """Enum to specify the type of stream to capture."""
+    STDOUT = 1
+    STDERR = 2
+
```
Comment (nichmor) [inline src/repror/internals/commands.py:23]: let's add 

@property
def is_stderr:
   return self == StreamType.STDOUT

File: src/repror/internals/commands.py [hunk 3]
```diff
@@ -8,32 +8,51 @@
...
+    with subprocess.Popen(command, cwd=cwd, env=env, **kwarwgs) as process:
+        # Print stderr as it comes in
+        for line in io.TextIOWrapper(
+            process.stderr, encoding="utf-8"
+        ):  # or another encoding
+            print(line, end="")
+
+        # Also print the other output if there is any
+        if stream_type == StreamType.STDERR:
+            out = process.stdout
```
Comment (nichmor) [inline src/repror/internals/commands.py:51]: out = process.stderr if stream_type.is_stderr else process.stdout

File: src/repror/internals/commands.py (hunk 1 above)
Comment (tdejager) [reply src/repror/internals/commands.py:48]: Changed it!

File: src/repror/internals/commands.py (hunk 2 above)
Comment (tdejager) [reply src/repror/internals/commands.py:23]: Changed it!

File: src/repror/internals/commands.py (hunk 3 above)
Comment (tdejager) [reply src/repror/internals/commands.py]: Changed it!

**Pull request merged**